### ISSUE TITLE
feature-request: Support multiple options in Function::CodeUri section

### ISSUE THREAD

**Issue by godrose** [aws/serverless-application-model#1746]
**Description:**

Need more options for specifying the code to be packaged before deployment

I have a case when there's certain replaceable part of the code which is required only for the certain configuration of the lambdas. I want to be able to omit if from the package so it never gets deployed.
This of course is relevant for the runtimes when there's no actual compilation artifacts, like `Node`

I was thinking about something like:

```yaml
Globals:
  Function:
    CodeUri: ./src |
      !./src/folder-i-want-to-omit
``` 

In short something like `.gitignore`. I would call this `.samignore` ;)

**Comment by jfuss** [comment]
This is actually a SAM CLI/Lambda Builders feature request not a SAM Spec. We have various requests for similar things already: https://github.com/aws/aws-lambda-builders/issues/185, https://github.com/aws/aws-sam-cli/issues/3350, https://github.com/aws/aws-sam-cli/issues/1360, and https://github.com/aws/aws-sam-cli/issues/858.

We likely need to combine those into a single one but will handle outside this issues.

Closing this as a duplicate of other requests in the correct repo (SAM CLI/Lambda builders).
